assets/js/app/Data.js

```diff
@@ -4,8 +4,60 @@
 
 "use strict";
 
+/* A named collection of data sources used to draw a plot*/
+LocusZoom.DataSources = function() {
+    this.sources = {};
+};
+
+LocusZoom.DataSources.prototype.addSource = function(ns, x) {
+    function findKnownSource(x) {
+        if (!LocusZoom.KnownDataSources) {return null;}
+        for(var i=0; i<LocusZoom.KnownDataSources.length; i++) {
+            if (!LocusZoom.KnownDataSources[i].SOURCE_NAME) {
+                throw("KnownDataSource at position " + i + " does not have a 'SOURCE_NAME' static property");
+            }
+            if (LocusZoom.KnownDataSources[i].SOURCE_NAME == x) {
+                return LocusZoom.KnownDataSources[i];
+            }
+        }
+        return null;
+    }
+
+    if (Array.isArray(x)) {
+        var dsclass = findKnownSource(x[0]);
+        if (dsclass) {
+            this.sources[ns] = new dsclass(x[1]);
+        } else {
+            throw("Unable to resolve " + x[0] + " data source");
+        }
+    } else {
+        this.sources[ns] = x;
+    }
+    return this;
+};
+
+LocusZoom.DataSources.prototype.getSource = function(ns) {
+    return this.sources[ns];
+};
+
+LocusZoom.DataSources.prototype.setSources = function(x) {
+    if (LocusZoom.isString(x)) {
```

Comment:
Clicking the [see note](https://github.com/statgen/locuszoom/pull/7/files#diff-8c4275047bc760f446e4203cd9f5e5fdR213) link should take you to the line in this pull req's diff where the `isString` function is defined (where our discussion of the function lives). I just threw this on all the instances of that functions usage that I could find.